plugins/modules/win_scheduledtask.ps1

```diff
@@ -0,0 +1,211 @@
+#!powershell
+
+# Copyright: (c) 2023, Yasmin Hinel (@yahikii)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+#AnsibleRequires -PowerShell Ansible.ModuleUtils.AddType
+
+
+
+$spec = @{
+    options             = @{
+        task_name   = @{ type = "str"; required = $true }
+        description = @{ type = "str" }
+        actions     = @{ type = "dict"; options = @{
+                arguments = @{ type = "str" }
+                execute   = @{ type = "str" }
+            }
+            required_if = @(
+                , @("actions", $true, @("arguments", "execute"))
+            )
+        }
+        triggers    = @{ type = "dict"; options = @{
+                start_time  = @{ type = "str" }
+                day_of_week = @{ type = "str"; choices = "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday" }
+                frequency   = @{ type = "str"; choices = "once", "daily", "weekly"; default = "once" }
+            }
+            required_if = @(
+                , @("triggers", $true, @("frequency", "start_time"))
+                , @("frequency", "weekly", @("day_of_week"))
+            )
+        }
+        username    = @{ type = "str" }
+        password    = @{ type = "str" }
```

Comment:
Missing "no_log" option for password. Otherwise the credentials are leaked in the windows eventlog.